src/python/pants/backend/python/dependency_inference/scripts/dependency_parser.py

```diff
@@ -18,13 +18,22 @@
 from io import open
 
 MIN_DOTS = os.environ["MIN_DOTS"]
+MIN_SLASHES = os.environ["MIN_SLASHES"]
 
 # This regex is used to infer imports from strings, e.g.
 #  `importlib.import_module("example.subdir.Foo")`.
 STRING_IMPORT_REGEX = re.compile(
     r"^([a-z_][a-z_\d]*\.){" + MIN_DOTS + r",}[a-zA-Z_]\w*$",
     re.UNICODE,
 )
+# This regex is used to infer resource names from strings, e.g.
+#  `load_resource("data/db1.json")
+# Since Unix allows basically anything for filenames, we require some "sane" subset of possibilities
+#  namely, word-character filenames and a mandatory extension.
+STRING_RESOURCE_REGEX = re.compile(
+    r"^([\w]*\/){" + MIN_SLASHES + r",}\w*(\.[^\/\.]+)+$",
+    re.UNICODE,
+)
```

Comment:
Open discussion: What's the harm in just matching every string?
- Potentially a little bit more of work to see if a string is a valid resource
- Over-matching. Is that really an issue for resources though? :thinking:

Reply:
There are a lot of string literals in code, so it might be a lot more work?

Reply:
We do the work in one place or another. In this case it is in matching regexes. But could be shifted into Pants.

Without numbers though, I'd believe the overhead is negligible.

The benefit is not scoping us out of extensionless files, or files in the same dir as the module.

Reply:
Another dimension here is I think users should be able to tweak `python-infer` at the `BUILD` level. So instead of dragging the overhead of string-matching-resources for the entire monorepo, it'd just be enabled for the specific files.

Reply:
I suppose we can also futher configure this with a `allow_extenionless` flag...